### DISCUSSION TITLE
UUID field validation

### DISCUSSION THREAD

**Discussion by beena-yatin-kanyal** [dropwizard/dropwizard#4780] [General]
We are using the dropwizard library for our microservices framework and would like to know if there is any support to validate the UUID fields. Consider the following DTO and the controller method attribute -

```
@JsonIgnoreProperties(ignoreUnknown = true)
public class SomeDto {
    @NotNull
    private final UUID id;
    @NotNull
    private final String type;

    public SomeDto(@JsonProperty("id") final UUID id,
                       @JsonProperty("type") final String type) {
        this.id = id;
        this.type = type;
    }

    public UUID getId() {
        return id;
    }
    
    . . . . .
```

_Calling mechanism_ -

```
public Response create(
          @PathParam("id") final UUID someId,
          @Valid final SomeDto someDto) throws Exception {
```

Please let us know if there is any valid support to address such a requirement.

**Comment by joschi** [comment]
@beena-yatin-kanyal The type of a parameter will be validated in the sense that the request will fail if the parameter `someId`  of `create(...)` cannot be deserialized into a UUID parameter.

**Comment by beena-yatin-kanyal** [comment]
@joschi - No.  Consider that if from some request I pass the invalid UUID we need custom validation that can validate the UUID format and throw a custom message. Is there any such support available?

**Comment by joschi** [comment]
@beena-yatin-kanyal You can write a custom validator or a custom `*Param` annotation.

- https://eclipse-ee4j.github.io/jersey.github.io/documentation/2.33/user-guide.html#d0e2273
- https://docs.oracle.com/javaee/7/api/javax/ws/rs/ext/ParamConverter.html
- https://github.com/dropwizard/dropwizard/blob/v2.0.20/dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/FuzzyEnumParamConverter.java

**Comment by beena-yatin-kanyal** [comment]
@joschi - I think it won't be a good idea to writing a param annotation to support multiple uuid's formats and would increase the code maintenance. Is there any support in the framework for such validation?? If not is there any milestone to have this added in dropwizard 2.x?

**Comment by joschi** [comment]
@beena-yatin-kanyal You can also validate the specific format of the `UUID` field in your constructor or in a `@JsonCreator` method.

There are no plans to implement something like this in Dropwizard specifically, since Jackson and JAX-RS/Jersey already provide this capability out of the box.
